numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2119,27 +2121,30 @@ array_empty_like(PyObject *NPY_UNUSED(ignored),
 
     if (npy_parse_arguments("empty_like", args, len_args, kwnames,
             "prototype", &PyArray_Converter, &prototype,
-            "|dtype", &PyArray_DescrConverter2, &dtype,
+            "|dtype", &PyArray_DTypeOrDescrConverterOptional, &dt_info,
             "|order", &PyArray_OrderConverter, &order,
             "|subok", &PyArray_PythonPyIntFromInt, &subok,
             "|shape", &PyArray_OptionalIntpConverter, &shape,
             NULL, NULL, NULL) < 0) {
         goto fail;
     }
-    /* steals the reference to dtype if it's not NULL */
-    ret = (PyArrayObject *)PyArray_NewLikeArrayWithShape(prototype, order, dtype,
-                                                         shape.len, shape.ptr, subok);
+    /* steals the reference to dt_info.descr if it's not NULL */
```

Comment:
I do prefer no stealing, but happy to leave it.

Reply:
I would also prefer not to steal the references, but `PyArray_NewLikeArrayWithShape` is called by `PyArray_NewLikeArray`, which has many consumers and I didn't want to go through all of them to make sure the references are correct.